core/lib/dal/src/models/storage_transaction.rs

```diff
@@ -364,12 +364,10 @@ impl From<StorageTransactionReceipt> for TransactionReceipt {
             to: storage_receipt
                 .transfer_to
                 .or(storage_receipt.execute_contract_address)
-                .map(|addr| {
-                    serde_json::from_value::<Address>(addr)
+                .and_then(|addr| {
+                    serde_json::from_value::<Option<Address>>(addr)
                         .expect("invalid address value in the database")
-                })
-                // For better compatibility with various clients, we never return null.
```

Comment:
Just to understand: Is this comment irrelevant now?

Reply:
No, removing it was probably a mistake. I changed it back to return the default address.